I-D_in_xml/draft-ietf-ccamp-optical-impairment-topology-yang-12.txt

```diff
@@ -0,0 +1,5320 @@
+        |  |  +--rw enabled?                     boolean
+        |  |  +--rw primary-path
+        |  |  |  +--rw network-ref?    leafref
+        |  |  |     ....................
+        |  |  +--rw backup-path* [index]
+        |  |  |  +--rw index           uint32
+        |  |  |  +--rw network-ref?    leafref
+        |  |  |     ....................
+        |  |  +--rw protection-type?             identityref
+
+   These attributes are used as follows:
+
+   *  "underlay": the presence of this container is indicating that an
+      underlying protection scheme exists
+   *  "enabled": (boolean) is set to 'true'
+   *  "primary-path": is referencing the primary OMS MCG (TE-link)
+   *  "backup-path": is referencing the secondary OMS MCG (TE-link)
+   *  "link-protection-type" (identityref) set to 'link-protection-1-
+      plus-1' as defined in [RFC8776]
+
+   The optical impairments for the underlying primary and secondary TE-
+   link can be described as for unprotected TE-links.  It may also be
+   possible to only describe the optical impairments for the 1+1
+   protected TE-link.  In this case the optical impairments of the worse
+   of the two underlying TE-links shall be used.  This should be
+   sufficient as input for path computation (worst case optical
+   feasibility consideration).
+
+   Any optical impairments imposed by the splitter and combiner
+   functions are included in the optical impairments of the ROADM
+   functions in the two adjacent WDM-TE-Nodes (connectivity matrix and
+   LLCL, respectively).
+
+
+
+Beller, et al.          Expires 14 September 2023              [Page 39]
+
+Internet-Draft    Opt. Impairment-Aware Topo YANG Model       March 2023
+
+
+2.10.2.2.  OMS MCG Protection Modeled as Single Protected TE-link
+
+   [[Editor's note: this sub-section is still work in progress]
+
+   This modeling approach abstracts the two physical links carrying the
+   same OMS MCG and the splitter/combiner functions in the two WDM-TE-
+   Nodes forming the protection group into a single TE-link.  If this
+   modeling approach is used the "te-link-attributes" already defined in
+   [RFC8795] and [RFC8776] are used as follows:
+
+   *  "underlay": this container must not be present
+   *  "link-protection-type" (identityref) set to 'link-protection-1-
+      plus-1' as defined in [RFC8776]
+
+   The optical impairments exposed for this 1+1 protected TE-link are
+   typically based on the optical impairments of the worse of the two
+   underlying physical links.  The optical impairments imposed by the
+   splitter and combiner functions can also be taken into account as
+   optical impairments of the 1+1 protected TE-link.  It may also be
+   possible to model the optical impairments imposed by the splitter and
+   combiner functions as contribution to the optical impairments of the
+   ROADM functions in the two adjacent WDM-TE-Nodes (connectivity matrix
+   and LLCL, respectively)
```

Comment:
The second part of the sentence is repeated: see comment above

The first part can in principle apply also to the case where the working and protectionTE-links are reported.

Initial straw man proposal for a common sentence at the end of section  2.10.2:

> The splitter/combiner functions are not modeled as separate entities in the model. Their optical impairments can be taken into account as optical impairments of the 1+1 protected TE-link or as optical impairments of the ROADM paths in the two adjacent WDM-TE-Nodes (connectivity matrix and LLCL, respectively).

While reviewing this text and the slides I have got the feeling that the consequences of which option is chosen can be described as:

> In the former case, the ROADM connectivity constraints and path impairments should be defined referencing the LTP terminating the 1+1 protected TE-link while in the latter case case, the ROADM connectivity constraints and path impairments should be defined referencing the LTP terminating the underlay working and protection TE-links.

What do you think?

Reply:
I think that the new sentence proposed is a well synthetic way to describe the two cases having the famous slides in front of the eye . The original one , apparently more verbose is more understandable even without having the slides. The sentence related "referencing the LTP...." should be in my view more elaborated to taken alone and not "in addition" to the original one.

Reply:
@sergiobelotti : do your comment refer to this sentence?

> In the former case, the ROADM connectivity constraints and path impairments should be defined referencing the LTP terminating the 1+1 protected TE-link while in the latter case case, the ROADM connectivity constraints and path impairments should be defined referencing the LTP terminating the underlay working and protection TE-links.

If so, IMHO, this sentence should be added to the previous one since it provides complimentary information to the first sentence.

In summary, the comment has three points:
1. describe the OI for splitter/combiner only once in section 2.10.2
2. some text proposal for the description:

    > The splitter/combiner functions are not modeled as separate entities in the model. Their optical impairments can be taken into account as optical impairments of the 1+1 protected TE-link or as optical impairments of the ROADM paths in the two adjacent WDM-TE-Nodes (connectivity matrix and LLCL, respectively).

4. consider whether the text proposed above is sufficient or whether it is worthwhile complementing it with the description of the consequences:

    > The splitter/combiner functions are not modeled as separate entities in the model. Their optical impairments can be taken into account as optical impairments of the 1+1 protected TE-link or as optical impairments of the ROADM paths in the two adjacent WDM-TE-Nodes (connectivity matrix and LLCL, respectively).
    > 
    > In the former case, the ROADM connectivity constraints and path impairments should be defined referencing the LTP terminating the 1+1 protected TE-link while in the latter case case, the ROADM connectivity constraints and path impairments should be defined referencing the LTP terminating the underlay working and protection TE-links.

Reply:
1. done - see https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/128#discussion_r1132751468
2. see text with some pure editorial changes: https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/128#discussion_r1132751468
3. Editor Notes added at the end of 2.10.2.1 and 2.10.2.2:
```
   [Editor's note: text and a figure based on the slide deck attached to
   Open Issue #119 on GitHub showing more details regarding the LTP of
   the 1+1 protected TE-link will be added here.]
```